jax/_src/sharding_impls.py

```diff
@@ -575,6 +576,7 @@ def shard_shape(self, global_shape: Shape) -> Shape:
           f'devices passed to PmapSharding. Got sharded dimension {sharded_dim} '
           f'with value {global_shape[sharded_dim]} in shape {global_shape} and '
           f'the number of devices={len(self._device_assignment)}')
+    assert sharded_shape is not None
```

Comment:
Why do you need this? Looks like this will always be True?

Can you just add a `else` branch to error after line 570 or silence pyright here?

Reply:
This is only True if `s` is guaranteed to either be `sharding_specs.{Unstacked,Chunk}`. Otherwise, it's undefined.

Reply:
Won't adding an else branch like I said above silence the error?